src/main/java/org/sopt/confeti/global/util/music/AppleMusicAPIHandler.java

```diff
@@ -295,4 +290,45 @@ private void validateFetchSize(int fetchSize) {
             throw new ConfetiException(ErrorMessage.BAD_REQUEST);
         }
     }
-}
+
+    @Override
+    @RetryOnTokenExpire
+    public List<ConfetiMusic> getMusicsByArtistIds(final Set<String> artistIds) {
+        if (artistIds.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        int totalArtists = artistIds.size();
+        List<String> artistIdList = new ArrayList<>(artistIds);
+        List<ConfetiMusic> resultMusics = new ArrayList<>();
+
+        if (totalArtists == 1) {
+            resultMusics.addAll(getTopSongsByArtistId(artistIdList.get(0), 3));
+        } else if (totalArtists == 2) {
+            resultMusics.addAll(getTopSongsByArtistId(artistIdList.get(0), 1));
+            resultMusics.addAll(getTopSongsByArtistId(artistIdList.get(1), 2));
+        } else {
+            for (int i = 0; i < Math.min(totalArtists, 3); i++) {
+                resultMusics.addAll(getTopSongsByArtistId(artistIdList.get(i), 1));
+            }
+        }
```

Comment:
로직 제안을 드리고 싶다는 생각을 했어요. 참고하셔서 리펙토링을 진행하셔도 좋고, 생각해두신게 있으시다면 그것으로 수정해주셔도 좋을 것 같아요.
다만, 하드 코딩되어있는 개수는 처리가 필요해보여요. `AppleMusicAPIHandler`는 프로젝트 전반적으로 사용되어야 해서 비즈니스적인 요구사항이 들어가는 것을 최대한 지양해서 구성했거든요. 그래서 리펙토링이 필요하다고 생각했습니다!

```suggestion
        if (totalArtists > PREFERENCE_PERFORMANCE_MUSIC_MAX_COUNT) { // 사실 에러 처리를 안해도 된다고 생각합니다. 비즈니스 로직이 포함된 Service나 Facade 계층에서 하는게 더 적절하다고 생각이 되네요.
            throw new ConfetiException(ErrorMessage.BAD_REQUEST);
        }

        for (int i = 0; i < totalArtists - 1; ++i) {
            resultMusics.add(getTopSongsByArtistId(artistIdList.get(i), 1));
        }

        resultMusics.addAll(getTopSongsByArtistId(artistList.get(totalArtists - 1), totalArtists - resultMusics.size()));

        // 마지막에 해당하는 아티스트가 나머지 음악 개수를 채우도록 구성하신 것 같아서 이렇게 작성해봤습니다.
        // 그런데, 마지막에 해당하는 아티스트의 음악 개수가 나머지 음악 개수를 충족하지 못한다면 어떨까요?
        // 그리고 1개만 추가했던 아티스트가 나머지 음악 개수를 충족할 수 있었다면 어떨까요?
        // 위 사항을 고려해서 리펙토링 해주시면 감사드리겠습니다.
```

Reply:
#346 에서 해당 로직을 PerformanceFacade 계층으로 이동했습니다. 
따라서, Facade 계층에서 하드 코딩된 기존 방식을 아래와 같이 수정했어요! 

1. 아티스트별로 MUSIC_FETCH_SIZE만큼 상위 곡 목록을 가져옴
2. round 인덱스를 기준으로 곡을 골라 추천 리스트를 채움

```java
 private List<ConfetiMusic> recommendMusicsByArtistCount(List<String> artistIdList, Set<String> existingMusicIds) {
    List<ConfetiMusic> musics = new ArrayList<>();
    Set<String> seenMusicIds = new HashSet<>(existingMusicIds);

    // 각 아티스트별 추천 곡 리스트를 round 인덱스로 순차적으로 탐색
    int round = 0;

    // 추천 곡이 원하는 개수(RECOMMEND_MUSIC_SIZE)만큼 모일 때까지 반복
    // 단, 무한 루프 방지를 위해 round가 MUSIC_FETCH_SIZE 이상이 되면 중단
    while (musics.size() < RECOMMEND_MUSIC_SIZE && round < MUSIC_FETCH_SIZE) {
        for (String artistId : artistIdList) {
            if (musics.size() >= RECOMMEND_MUSIC_SIZE) break;

            // 각 아티스트의 필터링된 상위 곡 리스트를 가져옴
            List<ConfetiMusic> topSongs = musicAPIHandler.getFilteredTopSongsByArtist(
                    artistId, MUSIC_FETCH_SIZE, seenMusicIds
            );

            // 현재 round 인덱스에 해당하는 곡이 있다면 추천 리스트에 추가
            if (round < topSongs.size()) {
                ConfetiMusic song = topSongs.get(round);
                if (!seenMusicIds.contains(song.getId())) {
                    musics.add(song);
                    seenMusicIds.add(song.getId());
                }
            }
        }
        round++;
    }

    return musics;
}
```